### PR TITLE
KAFKA-7147; ReassignPartitionsCommand should be able to connect to broker over SSL

### PR DESCRIPTION
**Pull request by lindong28**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by lindong28** [comment]
I will submit a small KIP later.

File: core/src/main/scala/kafka/admin/LogDirsCommand.scala
Comment (andrasbeni) [inline core/src/main/scala/kafka/admin/LogDirsCommand.scala:100]: @lindong28 the newly introduced config file might contain `bootstrap.servers=...`. Would it make sense to document that `--bootstrap-server` takes precedence? 
Have you thought of making `--bootstrap-server` optional and read servers from the config file? I know we are not doing this in other tools either, but I'm curious about your opinion.

File: core/src/main/scala/kafka/admin/LogDirsCommand.scala
Comment (omkreddy) [inline core/src/main/scala/kafka/admin/LogDirsCommand.scala]: Can we use props.putIfAbsent() for client_id config, so that users can overwrite client_id in props file.

File: core/src/main/scala/kafka/admin/LogDirsCommand.scala
Comment (lindong28) [reply core/src/main/scala/kafka/admin/LogDirsCommand.scala:100]: @andrasbeni Thanks for the suggestion. Yes it makes sense to document that `--bootstrap-server` takes precedence. 

Regarding making `--bootstrap-server ` optional, I think this approach also works just like the current solution which keeps `--bootstrap-server` required and `--bootstrap-server` take precedence. The former approach allows user the optional to only specify bootstrap server in the properties file whereas the latter doesn't. Sometimes flexibility is good but sometimes having multiple ways to do the same thing can also be confusing. Since I don't see a clear benefit for the extra flexibility, I would rather choose the easy way not to change whether `--bootstrap-server` is required. If anyone has a good reason to make it optional, I am OK with that as well.

File: core/src/main/scala/kafka/admin/LogDirsCommand.scala
Comment (lindong28) [reply core/src/main/scala/kafka/admin/LogDirsCommand.scala]: Thanks for the suggestion. I will make the change as suggested.

**Comment by omkreddy** [comment]
LGTM

**Comment by harshach** [comment]
+1 . @lindong28 do you want to write up a KIP before merge it in.

**Comment by lindong28** [comment]
@harshach Thanks for the review! The interface change in this patch has been approved in KIP-340. See https://cwiki.apache.org/confluence/display/KAFKA/KIP-340%3A+Allow+kafka-reassign-partitions.sh+and+kafka-log-dirs.sh+to+take+admin+client+property+file.

Previously I was waiting for review for this patch before committing it. A few days ago @omkreddy raised a comment that we may want to change the config name from `--config-file` to `--command-config`. The discussion of the config name is also related to KIP-332. Can you also comment which one you prefer?

**Comment by lindong28** [comment]
@harshach I have updated the patch to use `command-config` as the option name. Can you review again? Thanks.

**Comment by lindong28** [comment]
@becketqin Do you have time to review this patch? Thanks!

**Comment by harshach** [comment]
Thanks @lindong28 . +1.

**Review by harshach — APPROVED** [review]

**Comment by lindong28** [comment]
Thanks for review @andrasbeni @omkreddy @harshach. Merged to trunk.

**Pull request closed without merging**
